runtime/lua/vim/lsp/handlers.lua

```diff
@@ -206,6 +206,9 @@ M[ms.workspace_configuration] = function(_, result, ctx)
         value = vim.NIL
       end
       table.insert(response, value)
+    else
+      -- If no section is provided, return settings as is
```

Comment:
I think this is reasonable. However, I do believe that we should create an issue [upstream](https://github.com/microsoft/language-server-protocol) asking for this clarification to be made on the specs.

Reply:
+1, see also https://github.com/microsoft/language-server-protocol/issues/972